src/interfaces/coral_web/src/hooks/agents.ts

```diff
@@ -35,10 +35,46 @@ export const useCreateAgent = () => {
   });
 };
 
+export const useAgent = ({ agentId }: { agentId?: string }) => {
+  const cohereClient = useCohereClient();
+  return useQuery({
+    queryKey: ['agent', agentId],
+    enabled: !!agentId,
+    queryFn: async () => {
+      try {
+        return await cohereClient.getAgent(agentId ?? '');
+      } catch (e) {
+        console.error(e);
+        throw e;
+      }
+    },
+  });
+};
+
 /**
  * @description Returns a function to check if an agent name is unique.
  */
 export const useIsAgentNameUnique = () => {
   const { data: agents } = useListAgents();
-  return (name: string) => !agents?.some((agent) => agent.name === name);
+  return (name: string, omittedAgentId?: string) =>
+    !agents?.some((agent) => agent.name === name && agent.id !== omittedAgentId);
```

Comment:
a bit easier to parse since it can be easy to miss the `!`
```suggestion
    agents?.every((agent) => agent.name !== name || agent.id === omittedAgentId);
```